serialzy/serializers/base_model.py

```diff
@@ -118,3 +123,50 @@ def __check_module_fits_serializer(self, typ: Type) -> bool:
                 if base_name == self.module:
                     return True
         return False
+
+    def serialize_with_meta(self, obj: Any, dest_dir: str, model_file_name: str) -> None:
+        """
+        :param obj: object to serialize into bytes
+        :param dest_dir: files with serialized model and meta are written into destination directory
+        :param model_file_name: serialized model's file name
+        :return: None
+        """
+        dest_dir_path = Path(dest_dir)
+
+        if not dest_dir_path.is_dir():
+            raise NotADirectoryError
+        if not dest_dir_path.exists():
+            raise FileNotFoundError('directory does not exist')
+
+        typ = get_type(obj)
+        # check that obj type is valid
+        self._check_type(typ)
+
+        # write schema header
+        meta_file_path = dest_dir_path / ModelBaseSerializer.META_FILE_NAME
+        with open(meta_file_path, 'wb') as dest:
+            self._write_schema(typ, dest)
+
+        # write serialized model
+        model_file_path = dest_dir_path / model_file_name
+        with open(model_file_path, 'wb') as dest:
+            self._serialize(obj, dest)
+
+    def deserialize_with_meta(self, dest_dir: str, model_file_name: str, typ: Optional[Type] = None) -> Any:
+        """
+        :param dest_dir: directory with files with serialized model and meta
+        :param model_file_name: name of file with serialized model
+        :param typ: type of the resulting object, fetched from the meta if None
+        :return:
+        """
+        dest_dir_path = Path(dest_dir)
+
+        # read schema header
+        meta_file_path = dest_dir_path / ModelBaseSerializer.META_FILE_NAME
+        with open(meta_file_path, 'rb') as source:
```

Comment:
If you using pathlib, you should use it fully:
`with meta_file_path.open('rb') as source:`